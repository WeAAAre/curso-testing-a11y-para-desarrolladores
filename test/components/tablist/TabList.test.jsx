import { describe, it, expect } from 'vitest';
import { renderTabList } from '../../utils/renderTabList';

describe('TabList', () => {
  it('should match snapshot', async () => {
    const { container } = renderTabList({
      tabs: [
        {
          name: 'Primera tab',
          content: 'Contenido de la primera tab',
        },
        {
          name: 'Segunda tab (no seleccionada)',
          content: 'Contenido de la segunda tab',
        },
      ],
    });

    expect(container).toMatchSnapshot();
  });
});
