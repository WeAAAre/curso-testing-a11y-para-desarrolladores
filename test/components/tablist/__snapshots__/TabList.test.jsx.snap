// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabList > should match snapshot 1`] = `
<div>
  <div>
    <ul
      role="tablist"
    >
      <li
        aria-controls="Primeratab_panel"
        aria-selected="true"
        id="Primeratab_tab"
        role="tab"
        tabindex="0"
      >
        Primera tab
      </li>
      <li
        aria-controls="Segundatab(noseleccionada)_panel"
        aria-selected="false"
        id="Segundatab(noseleccionada)_tab"
        role="tab"
        tabindex="-1"
      >
        Segunda tab (no seleccionada)
      </li>
    </ul>
    <div
      aria-labelledby="Primeratab_tab"
      id="Primeratab_panel"
      role="tabpanel"
    >
      Contenido de la primera tab
    </div>
    <div
      aria-labelledby="Segundatab(noseleccionada)_tab"
      hidden=""
      id="Segundatab(noseleccionada)_panel"
      role="tabpanel"
    >
      Contenido de la segunda tab
    </div>
  </div>
</div>
`;
